File: smart-todo/src/ios/SmartTodo/Models/ActionStep.swift
import Foundation

/// Matches the API `ActionStep` shape exactly:
/// `{ id, todoId, title, description, order, isCompleted, createdAt }`.
///
/// Field names line up with the JSON keys, so no `CodingKeys` remapping is
/// required. `Identifiable` uses the server-generated `id`.
struct ActionStep: Codable, Identifiable, Equatable {
    let id: String
    let todoId: String
    var title: String
    var description: String
    let order: Int
    var isCompleted: Bool
    let createdAt: String
}

File: smart-todo/src/api/src/models/actionStep.ts
export interface ActionStep {
  id: string;
  todoId: string;
  title: string;
  description: string;
  order: number;
  isCompleted: boolean;
  createdAt: string;
}

export interface CreateActionStepInput {
  id: string;
  todoId: string;
  title: string;
  description: string;
  order: number;
}

export interface UpdateActionStepInput {
  isCompleted?: boolean;
}
